backend/dataall/cdkproxy/assets/gluedatabasecustomresource/index.py

```diff
@@ -25,6 +25,9 @@ def on_event(event, context):
 
 
 def on_create(event):
+    """Creates if it does not exist Glue database for the data.all Dataset
```

Comment:
just a note here for the future, using custom resource for glue db is not needed anymore, there is a construct for it: https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-glue-alpha.Database.html

Reply:
I think the issue has never been that the database construct did not exist, I think the main limitation is that the database does not come with the Lake Formation permissions. But I am seeing that this CFN resource exists now: https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_lakeformation/CfnPermissions.html. So a note for the future, we can think about replacing the custom resource by real resources!

Reply:
yep exactly, I think we can do something like this for setting up the privileges (as an example):
````
database_permissions = lf.CfnPermissions(
                self,
                f"GlueDatabasePermissions-{uuid.uuid4()}",
                data_lake_principal=lf.CfnPermissions.DataLakePrincipalProperty(data_lake_principal_identifier=role_arn),
                resource=lf.CfnPermissions.ResourceProperty(
                    database_resource=lf.CfnPermissions.DatabaseResourceProperty(catalog_id=catalog_id, name=glue_database.database_name),
                ),
                permissions=["CREATE_TABLE", "ALTER", "DROP"],
                permissions_with_grant_option=["CREATE_TABLE", "ALTER", "DROP"],
            )
            tables_permissions = lf.CfnPermissions(
                self,
                f"GlueDatabaseTablePermissions-{uuid.uuid4()}",
                data_lake_principal=lf.CfnPermissions.DataLakePrincipalProperty(data_lake_principal_identifier=role_arn),
                resource=lf.CfnPermissions.ResourceProperty(
                    table_resource=lf.CfnPermissions.TableResourceProperty(
                        catalog_id=catalog_id, database_name=glue_database.database_name, table_wildcard=lf.CfnPermissions.TableWildcardProperty()
                    ),
                ),
                permissions=["SELECT", "ALTER", "DESCRIBE"],
                permissions_with_grant_option=["SELECT", "ALTER", "DESCRIBE"],
            )
            database_permissions.node.add_dependency(glue_database)
            tables_permissions.node.add_dependency(glue_database)
````

Reply:
We are going to forget, so I opened an[ issue](https://github.com/awslabs/aws-dataall/issues/381). Maybe we can wait for the storage location CFN error to be fixed to replace both custom resources.

Reply:
Another motivation to avoid custom resources: I have done some "limitations" testing on data.all and I have reached a limitation on the number of datasets that can be created in an environment. The limitation comes from the custom resources Lambda policy size (check [this](https://repost.aws/knowledge-center/lambda-resource-based-policy-size-error)). I want to do some last minute changes to the custom resources providers in the dataset stack as part of this PR (needed for a customer).